landoapi/models/landing_job.py

```diff
@@ -0,0 +1,209 @@
+    # ordered, the resulting column will have the same order
+    # as the definition order of the enum. This can be relied
+    # on for comparisons or things like queries with ORDER BY.
+    status = db.Column(
+        db.Enum(LandingJobStatus), nullable=False, default=LandingJobStatus.SUBMITTED
+    )
+    created_at = db.Column(
+        db.DateTime(timezone=True), nullable=False, default=db.func.now()
+    )
+    updated_at = db.Column(
+        db.DateTime(timezone=True),
+        nullable=False,
+        default=db.func.now(),
+        onupdate=db.func.now(),
+    )
+
+    # JSON object mapping string revision id of the form "<int>" (used because
+    # json keys may not be integers) to integer diff id. This is used to
+    # record the diff id used with each revision and make searching for
+    # Transplants that match a set of revisions easy (such as those
+    # in a stack).
+    # e.g.
+    #     {
+    #         "1001": 1221,
+    #         "1002": 1246,
+    #         "1003": 1412
+    #     }
+    revision_to_diff_id = db.Column(JSONB, nullable=False)
+
+    # JSON array of string revision ids of the form "<int>" (used to match
+    # the string type of revision_to_diff_id keys) listing the order
+    # of the revisions in the request from most ancestral to most
+    # descendant.
+    # e.g.
+    #     ["1001", "1002", "1003"]
+    revision_order = db.Column(JSONB, nullable=False)
+
+    # Text describing errors when status != LANDED.
+    error = db.Column(db.Text(), default="")
+
+    # LDAP email of the user who requested transplant.
+    requester_email = db.Column(
+        db.String(254),
+        db.CheckConstraint("requester_email <> ''", name="requester_email_not_empty"),
+        nullable=False,
+    )
+
+    # Lando's name for the repository.
+    repository_name = db.Column(
+        db.Text(),
+        db.CheckConstraint("repository_name <> ''", name="repository_name_not_empty"),
+        nullable=False,
+    )
+
+    # URL of the repository revisions are to land to.
+    repository_url = db.Column(db.Text(), default="")
+
+    # Identifier for the most descendent commit created by this landing.
+    landed_commit_id = db.Column(db.Text(), default="")
+
+    # JSON list of bug ids tied to the revisions at creation time.
+    bug_ids = db.Column(JSONB, nullable=False)
+
+    # Number of attempts made to complete the job.
+    attempts = db.Column(db.Integer, nullable=False, default=0)
+
+    # Priority of the job. Higher values are processed first.
+    priority = db.Column(db.Integer, nullable=False, default=0)
+
+    def __repr__(self):
+        return "<LandingJob: %s>" % self.id
+
+    @property
+    def landing_path(self):
+        return [(int(r), self.revision_to_diff_id[r]) for r in self.revision_order]
+
+    @property
+    def head_revision(self):
+        """Human-readable representation of the branch head's Phabricator revision ID.
+        """
+        assert (
```

Comment:
Not sure if the use of `assert` here is ideal -- if we want to enforce this usage (outside of debugging), a specific exception should be raised instead. As an example, asserts are dropped in the bytecode if `-o` is used (see https://docs.python.org/3/using/cmdline.html#cmdoption-o)